### PR TITLE
Change some LuciBuildService methods to return Build objects

### PR DESCRIPTION
**Pull request by nehalvpatel**
`rescheduleBuild` and `rescheduleUsingCheckRunEvent` now return `Build` objects instead of `true`.

Addresses https://github.com/flutter/flutter/issues/83945

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: app_dart/lib/src/service/scheduler.dart [hunk 1]
```diff
@@ -450,7 +450,8 @@ class Scheduler {
             success = true;
           }
         } else {
-          success = await luciBuildService.rescheduleUsingCheckRunEvent(checkRunEvent);
+          await luciBuildService.rescheduleUsingCheckRunEvent(checkRunEvent);
```
Comment (ricardoamador) [inline app_dart/lib/src/service/scheduler.dart:453]: Are you going to do anything with the new return object?

File: app_dart/test/service/luci_build_service_test.dart [hunk 2]
```diff
@@ -740,12 +740,11 @@ void main() {
 
     test('Reschedule an existing build', () async {
       when(mockBuildBucketClient.scheduleBuild(any)).thenAnswer((_) async => generateBuild(1));
-      final bool rescheduled = await service.rescheduleBuild(
+      await service.rescheduleBuild(
```
Comment (ricardoamador) [inline app_dart/test/service/luci_build_service_test.dart:743]: I would add validation of the new return object here.

File: app_dart/lib/src/service/scheduler.dart (hunk 1 above)
Comment (nehalvpatel) [reply app_dart/lib/src/service/scheduler.dart:453]: Actually, I'm not sure. I'm addressing this issue as a first contribution and it doesn't mention that part: https://github.com/flutter/flutter/issues/83945

File: app_dart/test/service/luci_build_service_test.dart (hunk 2 above)
Comment (nehalvpatel) [reply app_dart/test/service/luci_build_service_test.dart:743]: Sounds good. I'll see what I can do.

File: app_dart/lib/src/service/scheduler.dart (hunk 1 above)
Comment (ricardoamador) [reply app_dart/lib/src/service/scheduler.dart:453]: Ah I see. Okay no problem.

**Review by ricardoamador — APPROVED** [review]
LGTM

File: app_dart/lib/src/service/luci_build_service.dart [hunk 3]
```diff
@@ -356,13 +356,12 @@ class LuciBuildService {
         ),
       ),
     );
-    return true;
   }
 
   /// Sends [ScheduleBuildRequest] for [pullRequest] using [checkRunEvent].
   ///
   /// Returns true if it is able to send the scheduleBuildRequest. Otherwise, false.
```
[the remark this answers is not in the document]
Comment (nehalvpatel) [reply app_dart/lib/src/service/luci_build_service.dart]: Done 👍

File: app_dart/lib/src/service/luci_build_service.dart [hunk 4]
```diff
@@ -326,7 +326,7 @@ class LuciBuildService {
   /// Sends [ScheduleBuildRequest] the buildset, user_agent, and
```
[the remark this answers is not in the document]
Comment (nehalvpatel) [reply app_dart/lib/src/service/luci_build_service.dart]: Done 👍

**Review by CaseyHillers — APPROVED** [review]
flutter-lgtm

**Comment by godofredoc** [comment]
@nehalvpatel can you please add  the `autosubmit` label to land this change?

**Comment by nehalvpatel** [comment]
Oh yea! I'll do that.

**Pull request merged**